### DISCUSSION TITLE
Prometheus Native Histograms in Mimir

### DISCUSSION THREAD

**Discussion by manishm** [grafana/mimir#3467] [Development]
When will the Prometheus Native Histograms in Mimir be merged into Mimir? 
This was recently merged in Prometheus - https://github.com/prometheus/prometheus/releases/tag/v2.40.0

**Comment by pracucci** [comment]
👋 Grafana team is working on it, but we don't have an ETA right now. @krajorama and @zenador are doing this work.

**Reply by manishm** [reply]
Thank you! Typically how long does it takes for Mimir to catch up with Prometheus updates?
Also - for native histograms do we think it might be available in Mimir sometime early next year? Is there any issue link that we can use to track?

**Reply by pracucci** [reply]
> Typically how long does it takes for Mimir to catch up with Prometheus updates?

We upgrade Prometheus in Mimir on a regular basis, but adding support for some new features (e.g. native histograms) in Mimir may take longer.

> Also - for native histograms do we think it might be available in Mimir sometime early next year? 

As I mentioned previously, I don't have an ETA to share, but it's something we're working on.

> Is there any issue link that we can use to track?

@krajorama @zenador do we? If not, can you open an issue to let community tracking the progress, please?

**Reply by krajorama** [reply]
Created this for tracking: https://github.com/grafana/mimir/issues/3478

**Comment by tonyhb** [comment]
Also, will this be enabled for the remote write API?  This has been enabled since ~July and improved with the native histogram addition in Oct/Nov.

Here's the "new" WriteRequest timeseries protobuf: https://github.com/prometheus/prometheus/blob/c0d8a56c69014279464c0e15d8bfb0e153af0dab/prompb/types.pb.go#L659.

If so, that'd be much more performant when doing remote writes to ingest data :)

**Reply by pracucci** [reply]
Yes, support for native histograms will include the remote write protocol support too.
